### DISCUSSION TITLE
How to log just the data without [[Handler]] [[Target]] etc?

### DISCUSSION THREAD

**Discussion by GorvGoyl** [pmndrs/valtio#918] [Q&A] [answered]
When I log a Valtio object, I get additional properties that make it difficult to see the actual object data. Is there a way to get just the data from a Valtio object for logging purposes?
 
![screenshot 2024-06-30 at 20 04 14@2x](https://github.com/pmndrs/valtio/assets/7106086/e0121f5e-ab27-4b4c-bdae-2266760c2c41)

**Comment by dai-shi** [comment]
Is this about vanilla proxy, or with `useSnapshot`? Or Both?

**Comment by GorvGoyl** [comment]
for proxy.

**Reply by dai-shi** [reply, marked as answer]
For proxies, I think the only valid solution is to use `snapshot`.
```js
import { proxy, snapshot } from 'valtio';

const state = proxy(...);
console.log(snapshot(state));
```
